core/src/main/java/org/opensearch/sql/planner/optimizer/rule/EvalPushDown.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.planner.optimizer.rule;
+
+import static org.opensearch.sql.planner.optimizer.pattern.Patterns.evalCapture;
+import static org.opensearch.sql.planner.optimizer.pattern.Patterns.limit;
+import static org.opensearch.sql.planner.optimizer.rule.EvalPushDown.EvalPushDownBuilder.match;
+
+import com.facebook.presto.matching.Capture;
+import com.facebook.presto.matching.Captures;
+import com.facebook.presto.matching.Pattern;
+import com.facebook.presto.matching.pattern.CapturePattern;
+import com.facebook.presto.matching.pattern.WithPattern;
+import java.util.List;
+import java.util.function.BiFunction;
+import lombok.Getter;
+import lombok.experimental.Accessors;
+import org.opensearch.sql.planner.logical.LogicalEval;
+import org.opensearch.sql.planner.logical.LogicalLimit;
+import org.opensearch.sql.planner.logical.LogicalPlan;
+import org.opensearch.sql.planner.optimizer.Rule;
+
+/** */
+public class EvalPushDown<T extends LogicalPlan> implements Rule<T> {
+
+  // TODO: Add more rules to push down sort and project
```

Comment:
Please add some integ-tests in `ExplainIT` for current rule.
Please provide your design details for discussion in https://github.com/opensearch-project/sql/issues/2864#issuecomment-2264635556 before going to next.

Reply:
I've updated more details in the OverView of that issue https://github.com/opensearch-project/sql/issues/2864#issue-2434840370

Reply:
It seems all tests in `ExplainIT` is about sql explain. Is there any sql syntax can produce eval operator now?

Reply:
> I've updated more details in the OverView of that issue [#2864 (comment)](https://github.com/opensearch-project/sql/issues/2864#issue-2434840370)

Could you create sub-issues for each task in your plan and link this PR to one of sub-issue?



> It seems all tests in `ExplainIT` is about sql explain. Is there any sql syntax can produce eval operator now?

I see. Could this rule effect the `size` parameter in `OpenSearchIndexScan`?  If yes, ExplainIT would be the proper one, or we should figure out how to verify the rule works in IT.

Reply:
> I see. Could this rule effect the size parameter in OpenSearchIndexScan? If yes, ExplainIT would be the proper one, or we should figure out how to verify the rule works in IT.

1. Sure, I'll add sub-issues for each one later.
2. Yes, it has such effect indirectly. There is another `ppl/ExplainIT` which could be the proper place to add an IT, I found the wrong one at first. I could add one. And an UT is already to verify it in `LogicalPlanOptimizerTest`.

Reply:
> I could add one. And an UT is already to verify it in LogicalPlanOptimizerTest.

I appreciate it. IT is required for any feature/enhancement, even critical bugfix.